test/integration/integration_test.cc

```diff
@@ -92,6 +92,39 @@ TEST_P(IntegrationTest, PerWorkerStatsAndBalancing) {
   check_listener_stats(0, 1);
 }
 
+// Validates that the drain actually drains the listeners.
```

Comment:
I'm sorry to be pedantic, but I still don't see how this test actually verifies that we have drained the listeners.

If the drain timeout is say 1s, could we somehow verify the listener is actually stopped at the end of the test? Either via a stat? Trying to connect and it failing? Getting a connection-close header on the response? Something?

Reply:
I have added a new stat and verified here. PTAL. Just to be clear on drain timeout, drain timeout does not have any impact here - listener would be stopped immediately on workers when `drain_listeners` endpoint is called.